pkg/repository/v1/sqlcv1/queue.sql

```diff
@@ -451,3 +451,30 @@ SELECT
     retry_count
 FROM ready_items
 RETURNING id, tenant_id, task_id, task_inserted_at, retry_count;
+
+-- name: CleanupV1QueueItem :execresult
+WITH qis as (
+    SELECT qi.task_id, qi.task_inserted_at, qi.retry_count
+    FROM v1_queue_item qi
+    WHERE NOT EXISTS (
+        SELECT 1
+        FROM v1_task vt
+        WHERE qi.task_id = vt.id
+            AND qi.task_inserted_at = vt.inserted_at
+    )
+    LIMIT @batchSize::int
+), locked_qis AS (
+    SELECT task_id, task_inserted_at, retry_count
+    FROM v1_queue_item
+    WHERE (task_id, task_inserted_at, retry_count) IN (
+        SELECT task_id, task_inserted_at, retry_count
+        FROM qis
+    )
+    order by id ASC
+    FOR UPDATE SKIP LOCKED
+)
```

Comment:
Sorry for being a PITA 😅 Can we simplify these two CTEs into a single one, or is there a reason for two? E.g. (untested):

```suggestion
WITH locked_qis as (
    SELECT qi.task_id, qi.task_inserted_at, qi.retry_count
    FROM v1_queue_item qi
    WHERE NOT EXISTS (
        SELECT 1
        FROM v1_task vt
        WHERE qi.task_id = vt.id
            AND qi.task_inserted_at = vt.inserted_at
    )
    ORDER BY id ASC
    LIMIT @batchSize::int
    FOR UPDATE SKIP LOCKED
)
```